packages/cli/src/models/network/NetworkController.ts

```diff
@@ -190,25 +190,18 @@ export default class NetworkController {
   // Contract model
   private _contractsListForPush(onlyChanged = false, changedLibraries: Contract[] = []): [string, Contract][] {
     const newVersion = this._newVersionRequired();
-    const pipeline = [
-      contracts => toPairs(contracts),
-      contracts =>
-        map(contracts, ([contractAlias, contractName]): [string, Contract] => [
-          contractAlias,
-          Contracts.getFromLocal(contractName),
-        ]),
-      contracts =>
-        filter(
-          contracts,
-          ([contractAlias, contract]) =>
-            newVersion ||
-            !onlyChanged ||
-            this.hasContractChanged(contractAlias, contract) ||
-            this._hasChangedLibraries(contract, changedLibraries),
-        ),
-    ];
-
-    return pipeline.reduce((xs, f) => f(xs), this.projectFile.contracts);
+
+    return toPairs(this.projectFile.contracts)
+      .map(
+        ([contractAlias, contractName]) => [contractAlias, Contracts.getFromLocal(contractName)] as [string, Contract],
```

Comment:
I think this line is not the same as above. With `([contractAlias, contractName]): [string, Contract] => [ ... ]`, you are telling the compiler the expected return value, and it will check that the implementation will conform to it. By changing it to `... as [string, Contract]`, you are manually casting it to what you want it to be, and losing the compiler type check.

Reply:
In fact we should tune linter to give warning on `as` usage and reduce them to zero eventually.